### DISCUSSION TITLE
Unable to clone repositry

### DISCUSSION THREAD

**Discussion by tysonstrange** [community/community#52379] [Repositories] [answered]
### Select Topic Area

Question

### Body

I made a new empty repo and trying to import data from an existing repeo.

When i use the GH website to import, or even if I clone the repo to my local desktop and then try to push it to my repo, it just hangs on writing objects. 

Trying to clone/import the data from this repo, but will not work. I think it's someting to do with how much data there is to push? How can I clone and push this data to my repo?

https://github.com/OTCv8/otcv8-dev

**Comment by Anmol-Baranwal** [comment]
It's possible that the repository you are trying to clone and push is quite large, which can cause the Git operation to become slow or unresponsive.

If you're having trouble pushing the repository from your local machine, you could try using Git's clone command instead. Here's how:

Open your terminal or Git Bash and navigate to the directory where you want to clone the repository.

Run the git clone command followed by the URL of the repository you want to clone, like so:

```git
git clone https://github.com/OTCv8/otcv8-dev.git
```

Copy
This will clone the entire repository to your local machine.

1. Once the repository has been cloned, navigate to the local directory, make any changes you need to make, and then commit your changes using `git commit -m "commit message"`.

2. Finally, push your changes to the remote repository using git push origin `master`.

If git push still hangs or becomes unresponsive, you could try increasing the Git buffer size by running `git config --global http.postBuffer 524288000`. This sets a larger buffer size, allowing Git to handle larger pushes more efficiently.

If none of these solutions work, you could try splitting the repository into smaller chunks or splitting it into several repositories. This can make it easier to manage and push your changes without running into size limitations.

**Reply by tysonstrange** [reply]
> It's possible that the repository you are trying to clone and push is quite large, which can cause the Git operation to become slow or unresponsive.
> 
> If you're having trouble pushing the repository from your local machine, you could try using Git's clone command instead. Here's how:
> 
> Open your terminal or Git Bash and navigate to the directory where you want to clone the repository.
> 
> Run the git clone command followed by the URL of the repository you want to clone, like so:
> 
> ```
> git clone https://github.com/OTCv8/otcv8-dev.git
> ```
> 
> Copy This will clone the entire repository to your local machine.
> 
> 1. Once the repository has been cloned, navigate to the local directory, make any changes you need to make, and then commit your changes using `git commit -m "commit message"`.
> 2. Finally, push your changes to the remote repository using git push origin `master`.
> 
> If git push still hangs or becomes unresponsive, you could try increasing the Git buffer size by running `git config --global http.postBuffer 524288000`. This sets a larger buffer size, allowing Git to handle larger pushes more efficiently.
> 
> If none of these solutions work, you could try splitting the repository into smaller chunks or splitting it into several repositories. This can make it easier to manage and push your changes without running into size limitations.



I wanted to clone it to keep all the commit history intact. 

I tried doing ```git config --global http.postBuffer 524288000``` then ran ```git push origin master``` but it just sits like this:

![image](https://user-images.githubusercontent.com/7628811/230847488-907b595f-79e0-4bc2-9a53-94aae6ad569e.png)


When i did the push from MSVS it eventually failed, but doing it from bash sat like this for over a day without progressing. 

Would it make any difference to clone, then checkout and old/early commit, then push from there, then checkout a newer/latest commit and push again? or does it make no difference in the amount of data it pushes.

Surely there are bigger repos around than this one, there must be a solution. 

Currently  trying again now and will see how it goes. When i check the .git folder there is a 2.5Gb .pack file. I have 100Mbs upload so shouldnt take that long.

**Reply by Anmol-Baranwal** [reply]
It's possible that the large size of the repository is causing the push operation to become slow or unresponsive. You could try splitting the repository into smaller chunks or splitting it into several repositories to make it easier to manage and push your changes without running into size limitations.

To answer your question, checking out an older commit and pushing from there can help minimize the amount of data being pushed, as only the changes made from that commit onwards will be pushed. However, keep in mind that this will not keep the entire commit history intact.

Additionally, git push can fail due to a variety of reasons such as connectivity issues, server issues, or authentication errors.

**Comment by tysonstrange** [comment]
I'vee been trying to clone and push it to my repo for several days now.

I tried checking out the initla commit to push that, but seems to have the same amount of objects being written. The plan was to push the initial ccommit, then checkout a later commit and push again to get the remaining commit history, but still hhavint the same problem. 

I want to keep all the cocmmit history without having to lose all the commit history.... is it possible to just squash some of the commit history or somthign to clear up some of the cache stored in the .pack file. maybe I can squash the commit hisory but still keep it.

**Comment by tysonstrange** [comment, marked as answer]
Ok, I finally figured it out.

Because the .pack file exceeded 2GB, which is bigger than the http.postBuffer can be, and also just exceeds the .pack file size even when using SSH, here's what I had to do.

I cloned the repo to local desktop, added the remote I wanted to push it to and then ran these git commands:
```
git push <remote name> <a very early commit ID>:refs/head/master
git push <remote name> <a later commit ID near the middle>:refs/head/master
git push <remote name> master
```
This caused the .pack file size to be smaller and within the size limit and allowed it to be pushed. The first push got the initial commit in, then about half the data, then the final push got the rest and it's cloned and up to date in my own repo.
